sim.py

```diff
@@ -6,69 +6,117 @@
 from scipy.spatial.transform import Rotation as R
 from time import time
 from simulation_manager import logger
+from FSW.controllers.controller import Controller
 import os
+import yaml
+from actuators.magnetorquer import Magnetorquer
+from actuators.reaction_wheels import ReactionWheel
+# from build.world.pyphysics.models import MagneticField, SRP
 
 START_TIME = time()
 
-controller_dt = 0.1
-estimator_dt = 1
-
-
 def controller(state_estimate):
-    return np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+    
+    return np.array([0, 0, 0, 0, 0, 0, 0])
 
 
 def estimator(y):
     return y
 
+
 def sensors(true_state): # TODO : sensors will return a shorter measurement vector and estimator will estimate the full state
-    return true_state + np.array([5000, 5000, 5000, 5, 5, 5, 0.05, 0.05, 0.05, 0.05, 2, 2, 2, 2])*np.random.random()
+    return true_state # + np.array([5000, 5000, 5000, 5, 5, 5, 0.05, 0.05, 0.05, 0.05, 2, 2, 2, 2])*np.random.random()
 
 
 def run(log_directory, config_path):
     logr = logger.MultiFileLogger(log_directory)
 
-    params = SimParams(config_path)
+    params = SimParams(config_path) # cpp params
+    with open(config_path, "r") as f:
+        pyparams = yaml.safe_load(f)
 
     initial_state = np.array(params.initial_state) # Get initial state
+    newstate = np.array(rk4(initial_state, np.zeros((params.num_MTBs + params.num_RWs, 1)), params, -params.dt, params.dt))
+    initial_state[13:19]  = newstate[13:19]
     num_RWs = params.num_RWs
     num_MTBs = params.num_MTBs
 
     # Logging Legend
     state_labels = ["r_x ECI [m]", "r_y ECI [m]", "r_z ECI [m]", "v_x ECI [m/s]", "v_y ECI [m/s]", "v_z ECI [m/s]",
-                "q_w", "q_x", "q_y", "q_z", "omega_x [rad/s]", "omega_y [rad/s]", "omega_z [rad/s]"] + ["omega_RW_" + str(i) + " [rad/s]" for i in range(num_RWs)]
+                "q_w", "q_x", "q_y", "q_z", "omega_x [rad/s]", "omega_y [rad/s]", "omega_z [rad/s]",
+                "rSun_x ECI [m]","rSun_y ECI [m]","rSun_z ECI [m]","xMag ECI [T]","yMag ECI [T]","zMag ECI [T]"] + ["omega_RW_" + str(i) + " [rad/s]" for i in range(num_RWs)]
     #measurement_labels = state_labels # TODO : Fix based on partial state measurement
     state_estimate_labels = ["r_hat_x ECI [m]", "r_hat_y ECI [m]", "r_hat_z ECI [m]", "v_hat_x ECI [m/s]", "v_hat_y ECI [m/s]", "v_hat_z ECI [m/s]",
-                "q_hat_w", "q_hat_x", "q_hat_y", "q_hat_z", "omega_hat_x [rad/s]", "omega_hat_y [rad/s]", "omega_hat_z [rad/s]"] + ["omega_hat_RW_" + str(i) + " [rad/s]" for i in range(num_RWs)]
-    input_labels = ["V_MTB_" + str(i) + " [V]" for i in range(num_MTBs)] + ["V_RW_" + str(i) + " [V]" for i in range(num_RWs)]
-
+                "q_hat_w", "q_hat_x", "q_hat_y", "q_hat_z", "omega_hat_x [rad/s]", "omega_hat_y [rad/s]", "omega_hat_z [rad/s]",
+                "rSun_x ECI [m]","rSun_y ECI [m]","rSun_z ECI [m]","xMag ECI [T]","yMag ECI [T]","zMag ECI [T]"] + ["omega_hat_RW_" + str(i) + " [rad/s]" for i in range(num_RWs)]
```

Comment:
Having Sun pos and Mag field as a part of the true state vector makes sense. Why is it also a part of the state estimate? I don't think FSW will be filtering the sun position and magnetometer readings

Reply:
we need a measurement vector, but ad hasnt been implemented. we can put them there.

Reply:
I will add a PR that adds sensor models after this one